src/generator/AutoRest.Go/Templates/ServiceClientTemplate.cshtml

```diff
@@ -28,49 +28,92 @@ import (
     {
-    if (!p.SerializedName.FixedValue.IsApiVersion())
-    {
-    @:@(string.Format((p.IsRequired || p.ModelType.CanBeEmpty()
-                                    ? "{0} {1}": "{0} *{1}"),
-                                    p.Name, p.ModelType.Name))
-    }
+        if (!p.SerializedName.FixedValue.IsApiVersion())
+        {
+            @:@(string.Format((p.IsRequired || p.ModelType.CanBeEmpty()
+                                                  ? "{0} {1}" : "{0} *{1}"),
+                                                  p.Name, p.ModelType.Name))
+        }
     }
 }
 
+
 @EmptyLine
 @WrapComment("// ", string.Format("New creates an instance of the {0} client.", Model.BaseClient))
 func New(@(Model.GlobalParameters))@(Model.BaseClient) {
-    return NewWithBaseURI(DefaultBaseURI, @(Model.HelperGlobalParameters))
+    @if (!Model.IsCustomBaseUri)
+    {
+        @:return NewWithBaseURI(DefaultBaseURI, @(Model.HelperGlobalParameters))
+    }
+    else
+    {
+        @:return NewWithoutDefaults(@(Model.HelperAllGlobalParameters))
+    }
 }
 
-@EmptyLine
-@WrapComment("// ", string.Format("NewWithBaseURI creates an instance of the {0} client.", Model.BaseClient))
-func NewWithBaseURI(baseURI string, @(Model.GlobalParameters)) @(Model.BaseClient) {
-   return @(Model.BaseClient){
-        Client: autorest.NewClientWithUserAgent(UserAgent()),
-        BaseURI: baseURI,
-        APIVersion: APIVersion,
-        @foreach (var p in Model.Properties)
-        {
-        if (!p.SerializedName.Value.IsApiVersion())
-        {
-        @:@(string.Format("{0}: {1},", p.Name, p.Name.Value.ToSentence()))
+@if (!Model.IsCustomBaseUri)
+{
+    <text>
+    @EmptyLine
+    @WrapComment("// ", string.Format("NewWithBaseURI creates an instance of the {0} client.", Model.BaseClient))
+    func NewWithBaseURI(baseURI string, @(Model.GlobalParameters)) @(Model.BaseClient) {
+        return @(Model.BaseClient){
+            Client: autorest.NewClientWithUserAgent(UserAgent()),
+            BaseURI: baseURI,
+            APIVersion: APIVersion,
+            @foreach (var p in Model.Properties)
+            {
+                if (!p.SerializedName.Value.IsApiVersion())
+                {
+                @:@(string.Format("{0}: {1},", p.Name, p.Name.Value.ToSentence()))
+                }
+            }
         }
+    }
+    </text>
+}
+else
+{
+    <text>
+    @EmptyLine
+    @WrapComment("// ", string.Format("NewWithoutDefaults creates an instance of the {0} client.", Model.BaseClient))
+    func NewWithoutDefaults(@(Model.AllGlobalParameters)) @(Model.BaseClient) {
+        return @(Model.BaseClient){
+            Client: autorest.NewClientWithUserAgent(UserAgent()),
+            APIVersion: APIVersion,
+            @foreach(var p in Model.Properties)
+            {
+                if (!p.SerializedName.Value.IsApiVersion())
+                {
+                    @:@(string.Format("{0}: {1},", p.Name, p.Name.Value.ToSentence()))
```

Comment:
Without digging in futher, is there any chance that this would need to be a pointer?